unified-codes/apps/web/src/components/templates/PropertyBrowser.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import { useParams } from 'react-router-dom';
+import { IEntity } from 'libs/data/src/lib';
+import { useState, useEffect } from 'react';
+import { Grid } from '@unified-codes/ui';
+
+export type PropertyBrowser = React.FunctionComponent;
+
+export const PropertyBrowser: PropertyBrowser = () => {
+  const { code } = useParams();
+  const url = `${process.env.NX_DATA_SERVICE_URL}:${process.env.NX_DATA_SERVICE_PORT}/${process.env.NX_DATA_SERVICE_GRAPHQL}`;
+  const [entity, setEntity] = useState<IEntity>();
+
+  useEffect(() => {
+    getEntity(url);
+  }, []);
+
+  const getEntityQuery = (code: string) => `
+  {
+    entity (code: "${code}") {
+      code
+      description
+      type
+      has_property {
+        type
+        value
+      }
+      # form_category
+      has_child {
+        code
+        description
+        type
+        # form
+        has_child {
+          code
+          description
+          type
+        }
+      }
+    }
+  }`;
+
+  const getEntity = async (url: string) => {
+    const response = await fetch(url, {
```

Comment:
this is all good - and I know that I said in my earlier comment to keep it local and not use redux..
On looking at it, I think it would be nice to use the redux saga and keep the `fetch` out of the component.

That way, you can dispatch an action in `getEntity` and just connect an `Entity` from the store to this component.

Either approach is fine - I like the idea of keeping the fetching in the same place, which would mean moving to the saga. and it simplifies this component, so that it is only concerned with display of data.

Reply:
Hehe, yeah I was very unsure about which was better. I guess it may be a good opportunity to fiddle with redux so I may move it for that reason as well as the benefit of a smaller component 👍